homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -308,23 +322,23 @@ def _calculate_new_probability(self):
 
         for obs in self.current_observations.values():
```

Comment:
I think here should also be `observation`
```suggestion
        for observation in self.current_observations.values():
```